### DISCUSSION TITLE
Config presets with private and public repositories

### DISCUSSION THREAD

**Discussion by jacek-jablonski** [renovatebot/renovate#17239] [Request Help] [answered]
Hi,
I'm a little confused about the organization of the config presets with private and public repositories.
In docs (https://docs.renovatebot.com/getting-started/private-packages/#private-config-presets), we can read:

> It's not recommended that you use a private repository to host your config while then extending it from a public repository.

Further, in the same paragraph:

> If your preset doesn't have secrets then you should make it public, while if it does have secrets then it's better to split your preset between a public one which all repos extend, and a private one with secrets which only other private repos extend.

So what would be the recommended approach if I have both private and public repositories? Is extending public config in private config forbidden? Should these configurations be disjoint in this situation?

**Comment by viceice** [comment, marked as answer]
- public repos should use public presets
- private repos should use public presets
- private and public repos can use private presets, but you should avoid them
- the hosted app can't use private presets from public repos
- the hosted app can only use private presets from same org for private repos

**Reply by jacek-jablonski** [reply]
For a clarification: why is it not recommended for private presets to extend from public presets?

**Reply by viceice** [reply]
i didn't said that
- private preset can extend public preset, but should be avoided
- public presets shouldn't extend private presets, as it's unsupported

**Reply by jacek-jablonski** [reply]
ok, I see. Many thanks for clarification.
